src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,227 @@
+                ["format"] = "JSON",
+                ["grant_type"] = "authorization_code",
+                ["method"] = "alipay.system.oauth.token",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture),
+                ["version"] = "1.0",
+            };
+            sortedParams.Add("sign", GetRSA2Signature(sortedParams, Options.GetPrivateKeyBytes()));
+
+            string address = QueryHelpers.AddQueryString(Options.TokenEndpoint, sortedParams!);
+
+            using var response = await Backchannel.GetAsync(address, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving an access token: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+            }
+
+            using var stream = await response.Content.ReadAsStreamAsync(Context.RequestAborted);
+            using var document = JsonDocument.Parse(stream);
+
+            var mainElement = document.RootElement.GetProperty("alipay_system_oauth_token_response");
+            if (!ValidateReturnCode(mainElement, out var code))
+            {
+                return OAuthTokenResponse.Failed(new Exception($"An error (Code:{code}) occurred while retrieving an access token."));
+            }
+
+            var payload = JsonDocument.Parse(mainElement.GetRawText());
+            return OAuthTokenResponse.Success(payload);
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            // See https://opendocs.alipay.com/apis/api_2/alipay.user.info.share for details.
+            var sortedParams = new SortedDictionary<string, string>()
+            {
+                ["app_id"] = Options.ClientId,
+                ["auth_token"] = tokens.AccessToken,
+                ["charset"] = "utf-8",
+                ["format"] = "JSON",
+                ["method"] = "alipay.user.info.share",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture),
+                ["version"] = "1.0",
+            };
+            sortedParams.Add("sign", GetRSA2Signature(sortedParams, Options.GetPrivateKeyBytes()));
```

Comment:
You always use the same key, so might as well fold it into the `GetRSA2Signature()` method.

```suggestion
            sortedParams.Add("sign", GetRSA2Signature(sortedParams));
```

Reply:
Done.